lightning_pose/utils/scripts.py

```diff
@@ -144,26 +144,26 @@ def get_data_module(
             "Update num_gpus to 1 in your config."
         )
     cfg.training.num_gpus = max(cfg.training.num_gpus, 1)
+    # Divide config batch_size by num_gpus to maintain the same effective batch
+    # size in a multi-gpu setting.
+    if cfg.training.train_batch_size % cfg.training.num_gpus != 0:
+        raise ValidationError(
+            f"train_batch_size should be a multiple of num_gpus. "
+            "train_batch_size={cfg.training.train_batch_size}, "
+            "num_gpus={cfg.training.num_gpus}"
+        )
+    if cfg.training.val_batch_size % cfg.training.num_gpus != 0:
+        raise ValidationError(
+            f"val_batch_size should be a multiple of num_gpus. "
+            "val_batch_size={cfg.training.val_batch_size}, "
+            "num_gpus={cfg.training.num_gpus}"
+        )
 
-    semi_supervised = check_if_semi_supervised(cfg.model.losses_to_use)
-    if not semi_supervised:
-        # Divide config batch_size by num_gpus to maintain the same effective batch
-        # size in a multi-gpu setting.
-        if cfg.training.train_batch_size % cfg.training.num_gpus != 0:
-            raise ValidationError(
-                f"train_batch_size should be a multiple of num_gpus. "
-                "train_batch_size={cfg.training.train_batch_size}, "
-                "num_gpus={cfg.training.num_gpus}"
-            )
-        if cfg.training.val_batch_size % cfg.training.num_gpus != 0:
-            raise ValidationError(
-                f"val_batch_size should be a multiple of num_gpus. "
-                "val_batch_size={cfg.training.val_batch_size}, "
-                "num_gpus={cfg.training.num_gpus}"
-            )
-        train_batch_size = int(cfg.training.train_batch_size / cfg.training.num_gpus)
-        val_batch_size = int(cfg.training.val_batch_size / cfg.training.num_gpus)
+    train_batch_size = int(cfg.training.train_batch_size / cfg.training.num_gpus)
+    val_batch_size = int(cfg.training.val_batch_size / cfg.training.num_gpus)
 
```

Comment:
update `test_batch_size` here, and pass to `BaseDataModule` and `UnlabeledDataModule` below

Reply:
Test batch size will not be split across GPUs so it doesn't need to be updated - unless I misunderstood your meaning